integration-tests/testutils/report/parser_test.go

```diff
@@ -24,88 +24,84 @@ import (
-	s.subject.Write([]byte(fmt.Sprintf("****** Running %s\n", s.testID)))
-
-	expected := fmt.Sprintf("test: %s\n", s.testID)
-	actual := s.output.String()
-
-	c.Assert(actual, check.Equals, expected,
-		check.Commentf("Expected text output %s not found, actual %s",
-			expected, actual))
+func (s *ParserReportSuite) SetUpTest(c *check.C) {
+	s.spy = &StatuserSpy{}
+	s.subject = &SubunitV2ParserReporter{statuser: s.spy}
 }
 
-func (s *ParserReportSuite) TestParserReporterReturnsTheNumberOfBytesWritten(c *check.C) {
-	actual, err := s.subject.Write([]byte(fmt.Sprintf("****** Running %s\n", s.testID)))
-	c.Assert(err, check.IsNil, check.Commentf("Error while writing to output %s", err))
-
-	expected := len([]byte(fmt.Sprintf("test: %s\n", s.testID)))
+func (s *ParserReportSuite) TestParserSendsNothingWitNotParseableInput(c *check.C) {
+	s.subject.Write([]byte("Not parseable"))
 
-	c.Assert(actual, check.Equals, expected,
-		check.Commentf("Expected length output %d not found, actual %d",
-			expected, actual))
+	c.Assert(len(s.spy.calls), check.Equals, 0,
+		check.Commentf("Unexpected event sent to subunit: %v", s.spy.calls))
 }
 
-func (s *ParserReportSuite) TestParserReporterOutputsSuccess(c *check.C) {
-	s.subject.Write([]byte(fmt.Sprintf("PASS: /tmp/snappy-tests-job/18811/src/github.com/ubuntu-core/snappy/integration-tests/tests/apt_test.go:34: %s      0.005s\n", s.testID)))
-
-	expected := fmt.Sprintf("success: %s\n", s.testID)
-	actual := s.output.String()
-
-	c.Assert(actual, check.Equals, expected,
-		check.Commentf("Expected text output %s not found, actual %s",
-			expected, actual))
+var eventTests = []struct {
+	gocheckOutput  string
+	expectedTestID string
+	expectedStatus string
+}{
+	{"****** Running testSuite.TestExists\n", "testSuite.TestExists", "exists"},
+	{"PASS: /tmp/snappy-tests-job/18811/src/github.com/ubuntu-core/snappy/integration-tests/tests/" +
+		"apt_test.go:34: testSuite.TestSuccess      0.005s\n", "testSuite.TestSuccess", "success"},
+	{"FAIL: /tmp/snappy-tests-job/710/src/github.com/ubuntu-core/snappy/integration-tests/tests/" +
+		"installFramework_test.go:85: testSuite.TestFail\n", "testSuite.TestFail", "fail"},
 }
 
-func (s *ParserReportSuite) TestParserReporterOutputsFailure(c *check.C) {
-	s.subject.Write([]byte(fmt.Sprintf("FAIL: /tmp/snappy-tests-job/710/src/github.com/ubuntu-core/snappy/integration-tests/tests/installFramework_test.go:85: %s\n", s.testID)))
-
-	expected := fmt.Sprintf("failure: %s\n", s.testID)
-	actual := s.output.String()
-
-	c.Assert(actual, check.Equals, expected,
-		check.Commentf("Expected text output %s not found, actual %s",
-			expected, actual))
+func (s *ParserReportSuite) TestParserReporterSendsEvents(c *check.C) {
+	for _, t := range eventTests {
+		s.spy.calls = []subunit.Event{}
+		s.subject.Write([]byte(t.gocheckOutput))
+
+		c.Check(len(s.spy.calls), check.Equals, 1,
+			check.Commentf("Unexpected events sent to subunit: %v", s.spy.calls))
```

Comment:
If you use check.HasLen, the Commentf is unnecessary as it will print the actual failed value in case of errors.